waffle/signals.py

```diff
@@ -8,4 +8,8 @@
 @receiver(m2m_changed, sender=get_waffle_flag_model().groups.through)
 def flag_membership_changed(sender, instance, action, **kwargs):
     if action in ('post_add', 'post_remove'):
-        instance.flush()
+        try:
+            instance.flush()
+        except AttributeError:
+            for flag in get_waffle_flag_model().objects.filter(pk__in=kwargs['pk_set']):
```

Comment:
Can you explain why this is necessary, please?  Is there anyway to do this without a database call?

Reply:
The problem here is `m2m_changed` handling from Django.

If i perform the following code
```
from my_app.models import MyUserModel
from waffle.models import Flag

flag = Flag.objects.first()
user = MyUserModel.objects.first()

flag.users.add(user) # all fine, cache is flushed without errors

user.flag_set.add(flag) # crash
```

the last line will crash because the `instance` received by `flag_membership_changed` is a `MyUserModel` instance instead of a `Flag` instance.

The only solution that i can see  when `instance` is of type user/group is to retrieve all the flag involved and flush them like i do here

It is not possible to guarantee that the flag involved are still accessible from `instance` when `flag_membership_changed` is executed.

Reply:
I have changed the code to use `reverse` instead of `try/except` to be more explicit of what is happening here

Reply:
Thanks for the explanation. Would you mind adding a comment with a version of this explanation, too?

Also, instead of checking for a `kwarg`, what if you checked: `if isinstance(instance, get_waffle_flag_model())`? That seems a bit safer than relying on the `kwarg`.

Reply:
I have applied the required changes.